lib/rspec/sidecar/services.rb

```diff
@@ -35,9 +39,9 @@ def service(name, type,
       loop do
         instances = zk.children(instances_path)
         if instances.empty?
-          raise "service '#{name}:#{type}' not available" if retries == 0
           retries -= 1
-          
+          raise SidecarNotAvailable, "service '#{name}:#{type}' not available" if retries == 0
```

Comment:
This was moved after the retries decrement because we were hitting zk one extra time as far as I could tell from my tests.